src/System.IO.Pipelines.Networking.Libuv/UvTcpConnection.cs

```diff
@@ -24,6 +22,7 @@ public class UvTcpConnection : IPipeConnection
 
         private Task _sendingTask;
         private WritableBuffer? _inputBuffer;
+        private MemoryHandle? _inputBufferPin;
```

Comment:
That you have had to implement this with a nullable type, makes we wonder if we should make Free callable several times, and only the first does something.